### PR TITLE
Change: @types/node dependency to dev dependency (In package.json)

### LINKED ISSUE #2965: Move types to devDependency.
https://github.com/ethereum/web3.js/blob/12361e5f04bac0df10e3dbd7a6d57a86677cf0d7/packages/web3/package.json#L64

Recommend moving `@types/node` to a `devDependency`.  It isn't needed at runtime, and if a library user is is targeting browser this dependency will give them incorrect type hints.

### PR DESCRIPTION
**Pull request by cokia**
## Description

moving @types/node to a devDependency. It isn't needed at runtime, and if a library user is is targeting browser this dependency will give them incorrect type hints.

Fixes #2965 

and  re confirmation request of #3337 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran ```npm run dtslint``` with success and extended the tests and types if necessary.
- [x] I ran ```npm run test:unit``` with success.
- [x] I have executed ``npm run test:cov`` and my test cases do cover all lines and branches of the added code.
- [x] I ran ```npm run build-all``` and tested the resulting file/'s from  ```dist``` folder in a browser.
- [] I have updated the ``CHANGELOG.md`` file in the root folder.
- [] I have tested my code on the live network.

### CODE REVIEW

**Review by cgewecke — COMMENTED** [review]
Just an initial review pass here but LGTM. The types tests [import `net`][1] from NodeJS and there's no other reference.  

@cokia Would it be possible to update package-lock.json too? You should be able to do this with:
```
npm uninstall --save-dev @types/node@12.6.1
npm install --save-dev @types/node@12.6.1
```



[1]: https://github.com/ethereum/web3.js/blob/a6ddec59e65116853435f203b25cb9c55824d084/packages/web3/types/tests/web3-test.ts#L21

**Comment by cokia** [comment]
@cgewecke i add Edit of package-lock.json :) 

@nivida  Please confirm this PR :D

File: packages/web3/package.json [hunk 1]
```diff
@@ -60,6 +59,7 @@
     },
     "devDependencies": {
         "definitelytyped-header-parser": "^1.0.1",
-        "dtslint": "0.4.2"
+        "dtslint": "0.4.2",
+	"@types/node": "^12.6.1"
```
Comment (holgerd77) [inline packages/web3/package.json:63]: The indentation here is wrong, can you make sure that you use spaces here and not tabs for consistency (please add this via rebase to the existing commit)?

**Comment by holgerd77** [comment]
Hi @cokia,
thanks for submitting this PR! 😄 Can you please rebase to update on top of the latest changes?

Regarding `package-lock.json`: I am not sure on this since I have no experience maintaining lock files in a rep (we don't add these on EthereumJS), @cgewecke these should likely only include the changes from the updated dependency here? Atm the changes on the file are huge.

**Comment by cgewecke** [comment]
@holgerd77 Yes, agree about the package-lock...will check this out locally and see what the minimal change set should be.

**Comment by cokia** [comment]
Should I change the tab to space and commit one more time?
I wonder if what you're saying is that you're turning a tab into a space.

**Comment by cgewecke** [comment]
@cokia Apologies, I will make the changes to this today if that's ok. Don't worry about the formatting.

**Comment by cgewecke** [comment]
@cokia I've re-opened this with the formatting fixes at #3453 because I didn't want to push changes directly to the default branch of your fork. Your authorship is preserved in the commit record. 

Closing here.

**Comment by cokia** [comment]
WFM. Thanks😊😊

**Pull request closed without merging**